scala/lambdas/ingest-flow-control/src/test/scala/uk/gov/nationalarchives/ingestflowcontrol/Helpers.scala

```diff
@@ -89,15 +89,19 @@ object Helpers {
     override def writeItems[T](tableName: String, items: List[T])(using format: DynamoFormat[T]): IO[List[BatchWriteItemResponse]] = notImplemented
 
     override def queryItems[U](tableName: String, requestCondition: RequestCondition, potentialGsiName: Option[String])(using returnTypeFormat: DynamoFormat[U]): IO[List[U]] =
-      notImplemented
-
-    override def getItems[T, K](primaryKeys: List[K], tableName: String)(using returnFormat: DynamoFormat[T], keyFormat: DynamoFormat[K]): IO[List[T]] =
-      val firstPK = primaryKeys.head.asInstanceOf[IngestQueuePartitionKey]
-      errors.raise(_.getItem, "Error getting item from dynamo table") >>
+      errors.raise(_.queryItem, "Error getting item from dynamo table") >>
         ref.get.map { existing =>
-          existing.filter(_.sourceSystem == firstPK.sourceSystem).sortBy(_.queuedAt).map(_.asInstanceOf[T])
+          (for {
+            values <- Option(requestCondition.attributes.values)
+            map <- values.toMap[String].toOption
```

Comment:
1. Could this not just be `map = values.toMap[String]`
2. The name `map` isn't too clear. When you convert `values` to a `Map`, what is the key and what is value `"S" -> "value"` or `"value" -> "value"` or is it still "attributeName" -> "value"?

Reply:
tbh, I do not know @MancunianSam what do you reckon?

Reply:
`toMap` is a method on `DynamoObject` which, despite it's name, returns `Either[DynamoReadError, Map[String, String]]` which is why this fiddling about with Options is necessary. (We could fiddle about with `Either` but `Option` is fine)

I think considering `requestCondition.attributes.values` is the line directly above, there's enough context for this to make sense so I'd say leave it as is.